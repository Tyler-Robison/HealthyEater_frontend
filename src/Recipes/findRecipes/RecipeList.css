

.RecipeList-container {
    display: flex;
    flex-direction: column;
    align-items: center;
    background-image: url('../../Images/food_background.jpg');
    background-repeat: no-repeat;
    background-size: cover;
    background-attachment: fixed;
}

.RecipeList-para {
    margin-top: 1rem;
    margin-bottom: 0.4rem;
}



.RecipeList-container button {
    margin-top: 1rem;
    box-shadow: 10px 5px 5px rgb(87, 87, 87);
}