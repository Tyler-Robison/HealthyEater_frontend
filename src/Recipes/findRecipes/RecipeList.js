import React, { useContext } from 'react';
import GlobalContext from '../../context/GlobalContext';
import { useNavigate } from "react-router-dom";
import Recipe from './Recipe';
import './RecipeList.css'

const RecipeList = () => {
    const { recipes } = useContext(GlobalContext)
    const navigate = useNavigate()

    console.log('recipes', recipes)

    return (
        <div className='RecipeList-container'>
            <button className='general-btn-red' onClick={() => navigate('/find_recipes')}>Go Back</button>
            {recipes.map(recipe => <Recipe recipe={recipe} />)}
            <button className='general-btn-red' onClick={() => navigate('/find_recipes')}>Go Back</button>
            {recipes.length === 0 && <p>Search for Recipes first</p>}
        </div>
    )
}

export default RecipeList;