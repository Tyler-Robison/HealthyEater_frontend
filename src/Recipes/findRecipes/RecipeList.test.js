import React, { useState } from 'react';
import { render, screen } from '@testing-library/react';
import RecipeList from './RecipeList';
import { MemoryRouter } from "react-router-dom";
import ContextProvider from '../../testContext'

// using listState to set 
const DummyComponent = () => {

    return (
        <RecipeList />
    )
}

it("renders without crashing", function () {
    <MemoryRouter>
        <ContextProvider >
            <DummyComponent />
        </ContextProvider>
    </MemoryRouter>
});

// recipeList provided in testContext as testRecipes
it("matches snapshot with filled recipeList", function () {

    const container = render(
        <MemoryRouter>
            <ContextProvider >
                <DummyComponent />
            </ContextProvider>
        </MemoryRouter>
    );
    expect(container.asFragment()).toMatchSnapshot();
});


it("matches snapshot with blank recipeList", function () {

    const container = render(
        <MemoryRouter>
            <ContextProvider initRecipes={[]}>
                <DummyComponent />
            </ContextProvider>
        </MemoryRouter>
    );
    expect(container.asFragment()).toMatchSnapshot();
});