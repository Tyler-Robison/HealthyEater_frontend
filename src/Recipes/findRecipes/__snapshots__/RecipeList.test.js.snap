// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot with blank recipeList 1`] = `
<DocumentFragment>
  <div
    class="RecipeList d-flex flex-column"
  >
    <div
      class="RecipeList-container"
    >
      <button
        class="general-btn-red"
      >
        Go Back
      </button>
      <button
        class="general-btn-red"
      >
        Go Back
      </button>
    </div>
  </div>
</DocumentFragment>
`;

exports[`matches snapshot with filled recipeList 1`] = `
<DocumentFragment>
  <div
    class="RecipeList d-flex flex-column"
  >
    <div
      class="RecipeList-container"
    >
      <button
        class="general-btn-red"
      >
        Go Back
      </button>
      <div
        class="row"
      >
        <div
          class="Recipe col-12"
        >
          <a
            class="RecipeList-para"
            href="/recipes/1234"
          >
            <b>
              Scrambled Eggs
            </b>
          </a>
          <p>
            Uses 2 of your ingredients ( eggs, otherIngredient )
          </p>
          <img
            alt="Picture of the recipe Scrambled Eggs"
            class="Recipe-img"
            src="https://spoonacular.com/recipeImages/716426-312x231.jpg"
          />
        </div>
      </div>
      <div
        class="row"
      >
        <div
          class="Recipe col-12"
        >
          <a
            class="RecipeList-para"
            href="/recipes/5678"
          >
            <b>
              BLT
            </b>
          </a>
          <p>
            Uses 3 of your ingredients ( bacon, lettuce, tomato )
          </p>
          <img
            alt="Picture of the recipe BLT"
            class="Recipe-img"
            src="https://spoonacular.com/recipeImages/716426-312x231.jpg"
          />
        </div>
      </div>
      <button
        class="general-btn-red"
      >
        Go Back
      </button>
    </div>
  </div>
</DocumentFragment>
`;
